src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -23,98 +23,87 @@ object SignedTransaction {
-
-  lazy val recoveredPublicKey: Option[Array[Byte]] = recoveredPointSign.flatMap { p =>
-    ECDSASignature.recoverPubBytes(
-      new BigInteger(1, signatureRandom.toArray[Byte]),
-      new BigInteger(1, signature.toArray[Byte]),
-      ECDSASignature.recIdFromSignatureV(p),
-      bytesToSign
-    )
   }
 
-  lazy val syntacticValidity: Boolean = {
+  def getSender(tx: Transaction, pointSign: Byte, signatureRandom: ByteString, signature: ByteString, recoveredPointSign: Option[Int]): Option[Address] = {
+    val bytesToSign: Array[Byte] = if (pointSign == negativePointSign || pointSign == positivePointSign) {
+      //global transaction
+      crypto.kec256(
+        rlpEncode(RLPList(
+          tx.nonce,
+          tx.gasPrice,
+          tx.gasLimit,
+          tx.receivingAddress.toArray,
+          tx.value,
+          tx.payload)))
+    } else {
+      //chain specific transaction
+      crypto.kec256(
+        rlpEncode(RLPList(
+          tx.nonce,
+          tx.gasPrice,
+          tx.gasLimit,
+          tx.receivingAddress.toArray,
+          tx.value,
+          tx.payload,
+          Config.Blockchain.chainId,
+          valueForEmptyR,
+          valueForEmptyS)))
+    }
 
-    import tx._
-    import Transaction._
+    val recoveredPublicKey: Option[Array[Byte]] = recoveredPointSign.flatMap { p =>
+      ECDSASignature.recoverPubBytes(
+        new BigInteger(1, signatureRandom.toArray[Byte]),
+        new BigInteger(1, signature.toArray[Byte]),
+        ECDSASignature.recIdFromSignatureV(p),
+        bytesToSign
+      )
+    }
 
-    def byteLength(b: BigInt): Int = b.toByteArray.length
+    for {
+      key <- recoveredPublicKey
+      addrBytes = crypto.kec256(key).slice(FirstByteOfAddress, LastByteOfAddress)
+      if addrBytes.length == Address.Length
+    } yield Address(addrBytes)
+  }
 
-    byteLength(nonce) <= NonceLength &&
-    //FIXME: the below doesn't make sense, an exception will be thrown if the condition is not met
-    //(receivingAddress.bytes.isEmpty || receivingAddress.bytes.length == AddressLength) &&
-    byteLength(gasLimit) <= GasLength &&
-    byteLength(gasPrice) <= GasLength &&
-    byteLength(value) <= ValueLength &&
-    signatureRandom.length <= ECDSASignature.RLength &&
-    signature.length <= ECDSASignature.SLength &&
-    recoveredSenderAddress.isDefined
+  def apply(tx: Transaction, pointSign: Byte, signatureRandom: ByteString, signature: ByteString): Option[SignedTransaction] = {
+    for {
+      recoveredPointSign <- SignedTransaction.getRecoveredPointSign(pointSign)
+      sender <- SignedTransaction.getSender(tx, pointSign, signatureRandom, signature, Some(recoveredPointSign))
+    } yield SignedTransaction(tx, pointSign, signatureRandom, signature, sender, recoveredPointSign)
   }
+}
+
+case class SignedTransaction (
+  tx: Transaction,
+  pointSign: Byte, //v
+  signatureRandom: ByteString, //r
+  signature: ByteString, //s
+  senderAddress: Address,
+  recoveredPointSign: Int) {
```

Comment:
where do we use `recoveredPointSign`?
Is it necessary if we already recovered public key and address?